src/shared/sol-iio.c

```diff
@@ -0,0 +1,990 @@
+        return false;
+    }
+
+    while (ent = readdir(dir), ent != NULL) {
+        if (strstr(ent->d_name, "trigger") == ent->d_name) {
+            if (check_trigger_name(ent->d_name, device->trigger_name)) {
+                device->manual_triggering = check_trigger_now_file(ent->d_name);
+                /* triggers dirs are of the form triggerX, so here we save X */
+                device->trigger_id = atoi(ent->d_name + strlen("trigger"));
+                return true;
+            }
+        }
+    }
+
+    return false;
+}
+
+static bool
+set_current_trigger(struct sol_iio_device *device, const char *trigger_name)
+{
+    char path[PATH_MAX + 1];
+    int len;
+
+    len = snprintf(path, sizeof(path), CURRENT_TRIGGER_DEVICE_PATH, device->device_id);
+    if (len < 0 || len > PATH_MAX) {
+        SOL_WRN("Could not open device current_trigger file");
+        return false;
+    }
+
+    if (!write_file(path, "%s", trigger_name)) {
+        SOL_WRN("Could not write to device current_trigger file");
+        return false;
+    }
+
+    return true;
+}
+
+static bool
+create_sysfs_trigger(struct sol_iio_device *device)
+{
+    int len, i;
+    struct stat st;
+    char *trigger_name;
+    bool result = false;
+
+    /* First, check if 'iio_sysfs_trigger' is available */
+    if (stat(SYSFS_TRIGGER_SYSFS_ADD_TRIGGER, &st)) {
+        SOL_WRN("No 'iio_sysfs_tigger' under '/sys/bus/iio/devices'."
+            " Missing 'modprobe iio-trig-sysfs'?");
+        return false;
+    }
+
+    /* TODO check available trigger indice instead of using a random one */
+    /* Create new trigger */
+    i = rand();
+    if (!write_file(SYSFS_TRIGGER_SYSFS_ADD_TRIGGER, "%d", i))
+        goto error;
+
+    /* Set device current trigger */
+    len = asprintf(&trigger_name, "sysfstrig%d", i);
+    if (len < 0)
+        goto error;
+    result = set_current_trigger(device, trigger_name);
+
+    if (result)
+        device->trigger_name = strdup(trigger_name);
+
+    free(trigger_name);
+
+    return result;
+
+error:
+    SOL_WRN("Could not create sysfs trigger.");
+
+    return false;
+}
+
+static bool
+check_trigger(struct sol_iio_device *device)
+{
+    char path[PATH_MAX + 1];
+    char *name = NULL;
+    int len;
+
+    len = snprintf(path, sizeof(path), CURRENT_TRIGGER_DEVICE_PATH, device->device_id);
+    if (len < 0 || len > PATH_MAX) {
+        SOL_WRN("Could not open IIO device%d trigger on sysfs", device->device_id);
+        return false;
+    }
+
+    len = sol_util_read_file(path, "%ms", &name);
+    if (len < 0) {
+        SOL_INF("No current trigger for iio:device%d. Creating a sysfs one.",
+            device->device_id);
+        if (!create_sysfs_trigger(device))
+            return false;
+    } else
+        device->trigger_name = name;
+
+    return check_manual_triggering(device);
+}
+
+static void
+set_buffer_size(struct sol_iio_device *device, int buffer_size)
+{
+    char path[PATH_MAX + 1];
```

Comment:
He does, since the snprintf() checks all around check if `len > PATH_MAX` instead of `len >= PATH_MAX`. Both are correct, but I prefer the one without `+ 1` since that's the one that's being used by the rest of the code.